### ISSUE TITLE
Incorrect codegen for self-FK

### ISSUE THREAD

**Issue by hiendaovinh** [stephenafamo/bob#23]
Given this schema, bob tests will fail
```
create table users (
        id serial primary key not null,
        email_validated  bool null default false,
-       primary_email    varchar(100) unique null
+       primary_email    varchar(100) unique null,
+       parent_id int,
+
+       foreign key (parent_id) references users (id)
 );
```

```
cleaning tables... DONE
migrating... DONE

-----------------
./users.go:454:2: rel declared but not used
 451| }
 452| 
 453| func (o *User) InsertParent(ctx context.Context, exec bob.Executor, related *UserSetter) error {
•454|   rel := related
 455| 
 456|   o.ParentID = omitnull.From(o.ID)
 457| 
 458|   inserted, err := UsersTable.Insert(ctx, exec, related)
-----------------

-----------------
./users.go:456:15: cannot use omitnull.From(o.ID) (value of type omitnull.Val[int]) as type null.Val[int] in assignment
 453| func (o *User) InsertParent(ctx context.Context, exec bob.Executor, related *UserSetter) error {
 454|   rel := related
 455| 
•456|   o.ParentID = omitnull.From(o.ID)
 457| 
 458|   inserted, err := UsersTable.Insert(ctx, exec, related)
 459|   if err != nil {
 460|           return fmt.Errorf("inserting related objects: %w", err)
-----------------
```

**Comment by stephenafamo** [comment]
I'll investigate

**Comment by stephenafamo** [comment]
BTW, thanks for such a detailed issue along with the test  🙏🏾 .

I would have liked you to add this change to the test as a PR so that you get credit in the git history of the product.

**Comment by stephenafamo** [comment]
Just pushed some changes to the tip that should make self-referencing foreign keys work. Try it out:

```sh
go get github.com/stephenafamo/bob@8dc3ae2
```

**Comment by stephenafamo** [comment]
Released in `v0.18.1`
